posts/2015-05-24-es6-gotchas.md

```diff
@@ -18,10 +18,29 @@ With ES6 components, there are a few "aha" moments and annoyances. This post was
 <div onClick={this.handleClick} />
 ~~~
 
-What's wrong with that? If you're not using `React.createClass`, which autobinds functions, you need to do it yourself. The correct way would be:
+What's wrong with that? If you're not using `React.createClass`, which autobinds functions, you need to do it yourself. The correct way would be add it to constructor:
 
 ~~~js
-<div onClick={this.handleClick.bind(this)} />
+import React from 'react'
+
+export default class ComponentName extends React.Component {
+  constructor(){
+    super()
+    this.handleClick = this.handleClick.bind(this)
+  }
+  ...
+~~~
+
+Another way of dealing with this is by enabling stage 0 in babel to include [es7 classProperties] (https://gist.github.com/jeffmo/054df782c05639da2adb)
+
+The idea is to combine fat arrow syntax which automatically binds to this.  A more detailed explanation can be found [here](http://www.ian-thomas.net/autobinding-react-and-es6-classes/)
+
+~~~js
+export default class ComponentName extends React.Component {
+  ...
+  handleClick = (e) => {
```

Comment:
This is good but risky advice, this is just a stage 0 proposal. Things can change dramatically between now and when this makes it in to the spec. IMO it should carry a warning that you may have to rewrite this code (perhaps multiple times) in the next year.